src/main/java/controller/AddUserServlet.java

```diff
@@ -0,0 +1,60 @@
+package controller;
+
+import factory.UserServiceFactory;
+import model.User;
+import org.apache.log4j.Logger;
+import service.UserService;
+import util.IdGeneratorUtil;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+
+@WebServlet(value = "/admin/add/user")
+public class AddUserServlet extends HttpServlet {
+
+    private static final UserService userService = UserServiceFactory.getUserService();
+    private static final Logger logger = Logger.getLogger(AddUserServlet.class);
+
+    @Override
+    protected void doGet(HttpServletRequest request,
+                         HttpServletResponse response) throws ServletException, IOException {
+        request.getRequestDispatcher("/add_user.jsp").forward(request, response);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request,
+                          HttpServletResponse response) throws IOException, ServletException {
+        final HttpSession session = request.getSession();
+        final String role = (String) session.getAttribute("role");
+        if (role.equals("admin")) {
```

Comment:
do we need to use this checking in all servlets? we use filter for this purpose. Filter will check all redirection to address "/admin/*"